Api/V8/AbstractApiController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+namespace Api\V8;
+
+use Api\V8\Exception\NotAllowedException;
+use Api\V8\Exception\NotFoundException;
+use Slim\Http\Response as HttpResponse;
+use Api\V8\Dictionary\Response;
+
+abstract class AbstractApiController
+{
+    /**
+     * @param HttpResponse $httpResponse
+     * @param int $status
+     * @param null $data
+     * @param string|null $message
+     * @return HttpResponse
+     */
+    public function generateResponse(
+        HttpResponse $httpResponse,
+        int $status,
+        $data = null,
+        string $message = null
+    ): HttpResponse {
```

Comment:
Before version 7.0 PHP never had facility to set return type. We have to make it php5 compatible?

Reply:
Let's talk about it.

Reply:
THe issue is that we still need to support PHP version 5.5.9 and above. Please see our compatibility matrix: https://suitecrm.com/wiki/index.php/Compatibility_Matrix#7.9.x